pkg/descheduler/policyconfig.go

```diff
@@ -46,10 +50,150 @@ func LoadPolicyConfig(policyConfigFile string) (*api.DeschedulerPolicy, error) {
 		return nil, fmt.Errorf("failed decoding descheduler's policy config %q: %v", policyConfigFile, err)
 	}
 
-	internalPolicy := &api.DeschedulerPolicy{}
-	if err := scheme.Scheme.Convert(versionedPolicy, internalPolicy, nil); err != nil {
+	// Build profiles
+	// TODO(jchaloup): replace this with v1alpha1 -> v1alpha2 conversion
+	internalPolicy, err := V1alpha1ToInternal(client, versionedPolicy)
+	if err != nil {
 		return nil, fmt.Errorf("failed converting versioned policy to internal policy version: %v", err)
 	}
 
 	return internalPolicy, nil
 }
+
+func V1alpha1ToInternal(
+	client clientset.Interface,
+	deschedulerPolicy *v1alpha1.DeschedulerPolicy,
+) (*api.DeschedulerPolicy, error) {
+	validStrategyNames := map[v1alpha1.StrategyName]interface{}{
+		"RemoveDuplicates":                            nil,
+		"LowNodeUtilization":                          nil,
+		"HighNodeUtilization":                         nil,
+		"RemovePodsViolatingInterPodAntiAffinity":     nil,
+		"RemovePodsViolatingNodeAffinity":             nil,
+		"RemovePodsViolatingNodeTaints":               nil,
+		"RemovePodsHavingTooManyRestarts":             nil,
+		"PodLifeTime":                                 nil,
+		"RemovePodsViolatingTopologySpreadConstraint": nil,
+		"RemoveFailedPods":                            nil,
+	}
+
+	var evictLocalStoragePods bool
+	if deschedulerPolicy.EvictLocalStoragePods != nil {
+		evictLocalStoragePods = *deschedulerPolicy.EvictLocalStoragePods
+	}
+
+	evictBarePods := false
+	if deschedulerPolicy.EvictFailedBarePods != nil {
+		evictBarePods = *deschedulerPolicy.EvictFailedBarePods
+		if evictBarePods {
+			klog.V(1).InfoS("Warning: EvictFailedBarePods is set to True. This could cause eviction of pods without ownerReferences.")
+		}
+	}
+
+	evictSystemCriticalPods := false
+	if deschedulerPolicy.EvictSystemCriticalPods != nil {
+		evictSystemCriticalPods = *deschedulerPolicy.EvictSystemCriticalPods
+		if evictSystemCriticalPods {
+			klog.V(1).InfoS("Warning: EvictSystemCriticalPods is set to True. This could cause eviction of Kubernetes system pods.")
+		}
+	}
+
+	ignorePvcPods := false
+	if deschedulerPolicy.IgnorePVCPods != nil {
+		ignorePvcPods = *deschedulerPolicy.IgnorePVCPods
+	}
+
+	var profiles []api.Profile
+
+	// Build profiles
+	for name, strategy := range deschedulerPolicy.Strategies {
+		if _, ok := validStrategyNames[name]; ok {
+			if strategy.Enabled {
```

Comment:
We are only generating profiles for enabled strategies here. I think this is fine, but I wanted to check if this is what we want to do. I notice while writing tests and not getting the disabled profiles populated

Reply:
Yes, only for enabled strategies for the moment. Logic for the list of disabled plugins will get implemented later.